tools/src/main/java/org/apache/kafka/tools/VerifiableConsumer.java

```diff
@@ -0,0 +1,609 @@
+                .type(Integer.class)
+                .setDefault(30000)
+                .metavar("TIMEOUT_MS")
+                .dest("sessionTimeout")
+                .help("Set the consumer's session timeout");
+
+        parser.addArgument("--verbose")
+                .action(store())
+                .required(false)
+                .setDefault(false)
+                .type(Boolean.class)
+                .metavar("VERBOSE")
+                .help("Set to false to only log rebalances and commits");
+
+        parser.addArgument("--enable-autocommit")
+                .action(store())
+                .required(false)
+                .setDefault(false)
+                .type(Boolean.class)
+                .metavar("ENABLE-AUTOCOMMIT")
+                .dest("useAutoCommit")
+                .help("Enable offset auto-commit on consumer");
+
+        parser.addArgument("--broker-list")
+                .action(store())
+                .required(true)
+                .type(String.class)
+                .metavar("HOST1:PORT1[,HOST2:PORT2[...]]")
+                .dest("brokerList")
+                .help("Comma-separated list of Kafka brokers in the form HOST1:PORT1,HOST2:PORT2,...");
+
+        parser.addArgument("--consumer.config")
+                .action(store())
+                .required(false)
+                .type(String.class)
+                .metavar("CONFIG_FILE")
+                .help("Consumer config properties file.");
+
+        return parser;
+    }
+
+    public static Properties loadProps(String filename) throws IOException {
+        Properties props = new Properties();
+        InputStream propStream = null;
+        try {
+            propStream = new FileInputStream(filename);
+            props.load(propStream);
+        } finally {
+            if (propStream != null)
+                propStream.close();
+        }
+        return props;
+    }
+
+    public static VerifiableConsumer createFromArgs(ArgumentParser parser, String[] args) throws ArgumentParserException {
+        Namespace res = parser.parseArgs(args);
+
+        String topic = res.getString("topic");
+        int maxMessages = res.getInt("maxMessages");
+        boolean useAutoCommit = res.getBoolean("useAutoCommit");
+        boolean verbose = res.getBoolean("verbose");
+        String configFile = res.getString("consumer.config");
+
+        Properties consumerProps = new Properties();
+        if (configFile != null) {
+            try {
+                consumerProps.putAll(loadProps(configFile));
+            } catch (IOException e) {
+                throw new ArgumentParserException(e.getMessage(), parser);
+            }
+        }
+
+        consumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, res.getString("groupId"));
+        consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, res.getString("brokerList"));
+        consumerProps.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, Boolean.toString(useAutoCommit));
```

Comment:
Since if useAutoCommit is turned on, we will always try to commit after each poll outside the consumer, so I feel we can safely set this config to false?